src/ProjectTemplates/ComponentsWebAssembly.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Program.cs

```diff
@@ -19,7 +21,17 @@ public static async Task Main(string[] args)
             var builder = WebAssemblyHostBuilder.CreateDefault(args);
             builder.RootComponents.Add<App>("app");
 
+#if (!Hosted || NoAuth)
             builder.Services.AddSingleton(new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+#else
+            builder.Services.AddScoped(sp =>
```

Comment:
This service registration needs to be scoped as all the services in `AddRemoteAuthentication` are registered as scoped. If we wanted to really make a singleton, we can check the runtime information on the registration code (in the extension methods) and conditionally register them as scoped/singleton based on the runtime.

In general we should "train" users to use scoped for max compatibility across client-side/server-side blazor.